.github/workflows/pr-merged.yml

```diff
@@ -73,16 +73,24 @@ jobs:
         id: checkout
         uses: actions/checkout@v4
         with:
-          sparse-checkout: flake/dev/diff-plugins.py
+          sparse-checkout: |
+            flake/dev/diff-plugins.py
+            ci/fetch-nixpkgs.nix
+            flake.lock
           sparse-checkout-cone-mode: false
 
       - name: Install Nix
         uses: cachix/install-nix-action@v31
         with:
-          nix_path: nixpkgs=channel:nixpkgs-unstable
           extra_nix_config: |
             accept-flake-config = true
 
+      - name: Fetch nixpkgs
+        run: |
+          nixpkgs=$(mktemp)
+          nix-instantiate --raw --eval ./ci/fetch-nixpkgs.nix > "$nixpkgs"
+          echo "NIX_PATH=nixpkgs=$(cat "$nixpkgs")" >> "$GITHUB_ENV"
```

Comment:
Looking at what @khaneliman just added to home-manager, we may be able to get rid of `fetch-nixpkgs.nix` entirely:

https://github.com/nix-community/home-manager/blob/7c455533409411b4053bb6d7db71eb35e0544254/.github/workflows/update-maintainers.yml#L48

Reply:
> I'm not sure what all the implications are (pros/cons), e.g. does not specifying the `narHash` have any major downsides for us in CI?

Hmm.. that's a good observation... I'm not sure if there's anything that could happen with a MITM in a github action runner or cause any issues that would give a false sense of security.

Reply:
I've gone with your approach for now.

Tested again here: https://github.com/MattSturgeon/nixvim/actions/runs/16035928139/job/45247572019?pr=24